### PR TITLE
Refactor prepare function

### PR DESCRIPTION
**Pull request by eikhr**
I refactored the promise code to use async/await syntax because it's a lot easier to read. I've also tried to use more understandable variable names.

### CODE REVIEW

File: src/prepare.js [hunk 1]
```diff
@@ -116,21 +107,26 @@ function prepareElement(element, errorHandler, context) {
...
     },
   } = options;
-  return prepareElement(element, errorHandler, context).then(
-    ([children, childContext, p]) =>
-      Promise.all(
-        React.Children.toArray(children)
-          .map((child) => prepare(child, options, childContext))
-          .concat(p)
-          .filter(Boolean),
-      ),
```
Comment (LudvigHz) [inline src/prepare.js:132]: tbh, I like this alot more. It's a bit more readable and understandable in my eyes.
Refactoring out the `prepareElement(...).then(` is alot better, but the rest is just creating unnecessary variables and control flow imo.

**Review by LudvigHz — APPROVED** [review]
Logic looks solid still.
The last refactoring is a _downgrade_ imo though.

File: src/prepare.js (hunk 1 above)
Comment (eikhr) [reply src/prepare.js:132]: I thought the part where it uses `.concat(p)` with a `p` that is possibly undefined, and then `.filter(Boolean)` to remove the (possible) undefined was a bit confusing. What do you think about this instead? 
```javascript
  await Promise.all(
    React.Children.toArray(children)
      .map((child) => prepare(child, options, childContext))
      .concat(preparePromise ?? Promise.resolve()),
  );
```

File: src/prepare.js [hunk 2]
```diff
@@ -116,20 +107,23 @@ function prepareElement(element, errorHandler, context) {
...
+    errorHandler,
+    context,
+  );
+
+  // Recursively run prepare on children, and return a promise that resolves once all children are prepared.
+  return Promise.all(
+    React.Children.toArray(children)
+      .map((child) => prepare(child, options, childContext))
+      .concat(preparePromise || []),
   );
```
Comment (eikhr) [inline src/prepare.js:127]: How's this @LudvigHz
I don't want to go in depth in a comment. It is a bad idea as comments are rarely maintained. This should point you in the right direction, and further understanding can be gained by reading the code.

**Pull request merged**